nucypher/utilities/metrics.py

```diff
@@ -1,23 +1,52 @@
 try:
-    from prometheus_client import Gauge, Enum, Counter, Info
+    from prometheus_client import Gauge, Enum, Counter, Info, Histogram, Summary
 except ImportError:
     raise ImportError('prometheus_client is not installed - Install it and try again.')
 from twisted.internet import reactor, task
 
 import nucypher
-from nucypher.blockchain.eth.agents import ContractAgency, StakingEscrowAgent
+from nucypher.blockchain.eth.agents import ContractAgency, StakingEscrowAgent, WorkLockAgent, PolicyManagerAgent
+from nucypher.blockchain.eth.actors import NucypherTokenActor
+from typing import List
+from nucypher.blockchain.eth.interfaces import BlockchainInterfaceFactory
 
-# Metrics
-known_nodes_guage = Gauge('known_nodes', 'Number of currently known nodes')
-work_orders_guage = Gauge('work_orders', 'Number of accepted work orders')
-missing_confirmation_guage = Gauge('missing_confirmations', 'Currently missed confirmations')
-learning_status = Enum('node_discovery', 'Learning loop status', states=['starting', 'running', 'stopped'])
-requests_counter = Counter('http_failures', 'HTTP Failures', ['method', 'endpoint'])
-host_info = Info('host_info', 'Description of info')
-active_stake_gauge = Gauge('active_stake', 'Active stake')
 
+class EventMetricsCollector:
 
-def collect_prometheus_metrics(ursula):
+    def __init__(self, staker_address, worker_address, contract_agent, event_name, argument_filters, metrics):
+        self.event_filter = contract_agent.contract.events[event_name].createFilter(fromBlock='latest',
+                                                                                    argument_filters=argument_filters)
+        self.metrics = metrics
+        self.event_name = event_name
+        self.staker_address = staker_address
+        self.worker_address = worker_address
+        self.contract_agent = contract_agent
+
+        if self.event_name == "ReStakeSet":
+            self.metrics["reStake"].set(self.contract_agent.is_restaking(staker_address))
+
+        if self.event_name == "WindDownSet":
```

Comment:
Is there a reason `ReStakeSet` and `WindDownSet` are checked here specifically during initialization and not as part of `collect`?